test/roundtrip_test.py

```diff
@@ -72,103 +78,141 @@
 A2rtAoGAMv92fqI+B5taxlZhTLAIaGVFbzoASHTRl3eQJbc4zc38U3Zbiy4deMEH
 3QTXq7nxWpE4YwHbgXAeJUGfUpE+nEZGMolj1Q0ueKuSstQg5p1nwhQIxej8EJW+
 7siqmOTZDKzieik7KVzaJ/U02Q186smezKIuAOYtT8VCf9UksJ4=
------END RSA PRIVATE KEY-----"""
+-----END RSA PRIVATE KEY-----""")
 
 
 class RoundtripTest(unittest.TestCase):
 
     def test_read_base64_key(self):
         key = rsa.RSAPublicKey(s)
-        self.assertEqual(key.fingerprint(), "\xfb\xa1\xeao\xd3y")
-        self.assertEqual(key.decoded, inner_s)
-        self.assertEqual(key.encoded[:15], "\x00\x00\x00\x07ssh-rsa"
-                                           "\x00\x00\x00\x01")
+        self.assertEqual(key.fingerprint(), six.b("\xfb\xa1\xeao\xd3y"))
+        self.assertEqual(key.decoded, six.b(inner_s))
+        self.assertEqual(
+            key.encoded[:15],
+            six.b("\x00\x00\x00\x07ssh-rsa\x00\x00\x00\x01")
+        )
 
     def test_read_binary_key(self):
-        key = rsa.RSAPublicKey(ssh.base64url_decode(s.split(" ")[1]))
-        self.assertEqual(key.fingerprint(), "\xfb\xa1\xeao\xd3y")
-        self.assertEqual(key.decoded, inner_s)
-        self.assertEqual(key.encoded[:15], "\x00\x00\x00\x07ssh-rsa"
-                                           "\x00\x00\x00\x01")
+        key = rsa.RSAPublicKey(ssh.base64url_decode(
+            s.split(six.b(" "))[1])
+        )
+        self.assertEqual(key.fingerprint(), six.b("\xfb\xa1\xeao\xd3y"))
+        self.assertEqual(key.decoded, six.b(inner_s))
+        self.assertEqual(
+            key.encoded[:15],
+            six.b("\x00\x00\x00\x07ssh-rsa\x00\x00\x00\x01")
+        )
 
     def test_create_challenge(self):
-        auth_server = server.AuthServer("gurka", DummyKeyProvider(),
-                                        "server.name")
-        s = auth_server.create_challenge("noa")
+        auth_server = server.AuthServer(six.b("gurka"), DummyKeyProvider(),
```

Comment:
These should also be strings (passwords, and server names are strings not arbitrary bytes right?)